src/k8s/cmd/k8s/k8s_x_snapd_config.go

```diff
@@ -0,0 +1,99 @@
+package k8s
+
+import (
+	apiv1 "github.com/canonical/k8s/api/v1"
+	cmdutil "github.com/canonical/k8s/cmd/util"
+	"github.com/canonical/k8s/pkg/utils/experimental/snapdconfig"
+	"github.com/spf13/cobra"
+)
+
+func newXSnapdConfigCmd(env cmdutil.ExecutionEnvironment) *cobra.Command {
+	disableCmd := &cobra.Command{
+		Use:   "disable",
+		Short: "Disable the use of snap get/set to manage the cluster configuration",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := snapdconfig.Disable(cmd.Context(), env.Snap); err != nil {
+				cmd.PrintErrf("Error: failed to disable snapd configuration: %v\n", err)
+				env.Exit(1)
+			}
+		},
+	}
+	reconcileCmd := &cobra.Command{
+		Use:   "reconcile",
+		Short: "Reconcile the cluster configuration changes from k8s {set,get} <-> snap {set,get} k8s",
+		Run: func(cmd *cobra.Command, args []string) {
+			mode, empty, err := snapdconfig.ParseMeta(cmd.Context(), env.Snap)
+			if err != nil {
+				if !empty {
+					cmd.PrintErrf("Error: failed to parse meta configuration: %v\n", err)
+					env.Exit(1)
+					return
+				}
+
+				cmd.PrintErrf("Warning: failed to parse meta configuration: %v\n", err)
+				cmd.PrintErrf("Warning: ignoring further errors to prevent infinite loop\n")
+				if setErr := snapdconfig.SetMeta(cmd.Context(), env.Snap, snapdconfig.Meta{
+					APIVersion: "1.30",
```

Comment:
When we bump the kubernetes version, will we have to remember to change this value?

Reply:
This is the "API version" of the snapd configuration format. It is named like this to note that it originates with the 1.30 release. We do not need to bump this on each release. Rather, this should be bumped whenever breaking changes happen (e.g. a configuration field changes, or a new version of snapd comes around).